### ISSUE TITLE
Fool model toward specified target class using FSGM

### ISSUE THREAD

**Issue by yenchenlin** [openai/cleverhans#52]
Hello @goodfeli and @npapernot ,

In my understanding, [these two lines](https://github.com/openai/cleverhans/blob/master/cleverhans/attacks_tf.py#L35-L36) get the result of what trained model is gonna classify the clean example. 

Then by adding this gradient to the image, the trained model will tend to not classify it as the same label as before.

However, I'm wondering how can I use FSGM to fool model so that it will classify all examples to a specified label, e.g., 0.

**Comment by yenchenlin** [comment]
```python
def fgsm_to_class_0(x, predictions, eps, clip_min=None, clip_max=None):

    # Compute loss
    zero = tf.constant(0, dtype=tf.float32)
    mask = tf.convert_to_tensor([1, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=tf.float32)
    y = tf.to_float(tf.not_equal(predictions*mask, zero))
    y = y / tf.reduce_sum(y, 1, keep_dims=True)
    loss = utils_tf.model_loss(y, predictions, mean=False)

    # Define gradient of loss wrt input
    grad, = tf.gradients(loss, x)

    # Take sign of gradient
    signed_grad = tf.sign(grad) * -1

    # ... same as original fgsm function
```

I've tried using above code to raise the score of particular class's logits, but it didn't work 😢

**Comment by yenchenlin** [comment]
Sorry, I figured out how to make it work.
